File: src/main/java/com/example/airmed/Entity/SocialContext.java
package com.example.airmed.Entity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class SocialContext {
    @Id
    @GeneratedValue(strategy =  GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, updatable = false)
    private Long id;

    @Column(nullable = false)
    private String occupation;

    @Column(nullable = false)
    private String highestEdu;

    @Column(nullable = false)
    private String relationship;

    @Column(nullable = false)
    private boolean sexuallyActive;

    @Column(nullable = false)
    private String genderOrientatin;

    @Column(nullable = false)
    private String legalProblems;

    @Column(columnDefinition = "boolean default false")
    private boolean adopted;

    @Column
    private String family;

    @Column
    private String familyPsychiatric;

    @Column
    private String drugs;

    @Column
    private String alcohol;

    @Column
    private String abuseMeds;

    @Column
    private String caffeine;

    @Column
    private String cigarettes;

    @Column
    private String exercise;

    @ManyToOne
    @JoinColumn(name = "patient", nullable = false)
    private Patient patient;
}

File: frontend/src/classes/SocialContext.ts
import {Patient} from "./Patient.ts";

export class SocialContext{
    id: number;
    occupation: string;
    highestEdu: string;
    relationship: string;
    sexuallyActive: boolean;
    genderOrientatin: string;
    legalProblems: string;
    adopted: boolean;
    family: string;
    familyPsychiatric: string;
    drugs: string;
    alcohol: string;
    abuseMeds: string;
    caffeine: string;
    exercise: string;
    patient: Patient;


    constructor(id: number, occupation: string, highestEdu: string, relationship: string, sexuallyActive: boolean, genderOrientation: string, legalProblems: string, adopted: boolean, family: string, familyPsychiatric: string, drugs: string, alcohol: string, abuseMeds: string, caffeine: string, exercise: string, patient: Patient) {
        this.id = id;
        this.occupation = occupation;
        this.highestEdu = highestEdu;
        this.relationship = relationship;
        this.sexuallyActive = sexuallyActive;
        this.genderOrientatin = genderOrientation;
        this.legalProblems = legalProblems;
        this.adopted = adopted;
        this.family = family;
        this.familyPsychiatric = familyPsychiatric;
        this.drugs = drugs;
        this.alcohol = alcohol;
        this.abuseMeds = abuseMeds;
        this.caffeine = caffeine;
        this.exercise = exercise;

        this.patient = new Patient(patient.id,patient.pnc,patient.firstName,
            patient.lastName,patient.mail, patient.phone, patient.psychiatrist,
            patient.psychotherapist
        );
    }
}